apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx

```diff
@@ -0,0 +1,251 @@
+import { ChangeEvent, FocusEvent, KeyboardEvent, useRef } from "react";
+import {
+  Box,
+  Text,
+  Grid,
+  IconButton,
+  TextField,
+  Tooltip,
+  IconButtonWithMenu,
+  numericScrubControl,
+} from "@webstudio-is/design-system";
+import { getFinalValue } from "../../shared/get-final-value";
+import { useIsFromCurrentBreakpoint } from "../../shared/use-is-from-current-breakpoint";
+import { ControlProps } from "../../style-sections";
+import { iconConfigs, StyleConfig } from "../../shared/configs";
+import { type StyleValue, type Unit, units } from "@webstudio-is/react-sdk";
+import { ChevronDownIcon } from "@webstudio-is/icons";
+import { PropertyName } from "../../shared/property-name";
+
+const commonUnitsSubset = units
+  .slice(0)
+  .sort((v) =>
+    ["%", "px", "rem", "em", "ch", "vh", "vw", "hv", "vmin", "vmax"].includes(v)
+      ? -1
+      : 1
+  );
+
+const CSSGridLeft = {
+  "& ~ input": {
+    paddingLeft: "calc($sizes$6 + 6px)",
```

Comment:
why do we have 6px, you sure we couldn't have used one of the design system values?